docs/Development-Guides/Rate-Limits.md

```diff
@@ -29,3 +22,166 @@ Product | Max requests initiated per minute
+1. Token B makes `5` requests to the `company` endpoint, `3` requests to the `directory` endpoint, and `2` requests to the `payment` endpoint within the same minute. Token B’s first 4 requests to the `company` endpoint will succeed, but the 5th will fail and return a 429 rate limit error. Note: Only succeeded requests count towards the application level rate limits.
+    - Access Token B rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 4/4 (FULL)
+        `directory` | 3/4
+        `individual` | 0/4
+        `employment` | 0/4
+        `payment` | 2/2 (FULL)
+        `pay-statement` | 0/2
+    - Application-level rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 8/20
+        `directory` | 6/20
+        `individual` | 0/20
+        `employment` | 0/20
+        `payment` | 4/12
+        `pay-statement` | 0/12
+1. Token C, D, and E repeats the same process as Token B making `5` requests to the `company` endpoint, `3` requests to the `directory` endpoint, and `2` requests to the `payment` endpoint all within the same minute. Similarly, every 5th request to the `company` endpoint for each token fails and returns a 429 rate limit error.
+    - Access Token C, D, and E individual rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 4/4 (FULL)
+        `directory` | 3/4
+        `individual` | 0/4
+        `employment` | 0/4
+        `payment` | 2/2 (FULL)
+        `pay-statement` | 0/2
+    - Application-level rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 20/20 (FULL)
+        `directory` | 15/20
+        `individual` | 0/20
+        `employment` | 0/20
+        `payment` | 12/12 (FULL)
+        `pay-statement` | 0/12
+1. If Token F makes `1` request to the `company` endpoint and `1` request to the `directory` endpoint. The `company` request will fail and return a 429 rate limit error even though Token F has not maxed out its access-token specific `company` bucket. That is because the application-level bucket is now full. No requests from any token will succeed calling the `company` endpoint until the `company` bucket’s 60-second Time-To-Live (TTL) timer resets. However, Token F’s request to the `directory` endpoint will succeed because its access-token specific `directory` bucket is not full AND the application-level `directory` bucket is not full either.
+    - Access Token F rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 1/4
+        `directory` | 1/4
+        `individual` | 0/4
+        `employment` | 0/4
+        `payment` | 0/2
+        `pay-statement` | 0/2
+    - Application-level rate limits
+        Bucket | Capacity
+        -------|-------------
+        `company` | 20/20 (FULL)
+        `directory` | 16/20
+        `individual` | 0/20
+        `employment` | 0/20
+        `payment` | 12/12 (FULL)
+        `pay-statement` | 0/12
```

Comment:
This is very long-winded and more of a step-by-step example rather than a doc. Should this instead go on a new page?

Reply:
Also @joshuakrohn an image may really really help us convey this information here.

Reply:
I agree that an image would be better, but working with what we have at the moment.

Reply:
@logangingerich where do you think this could go instead?

Reply:
I had been thinking about it being a best practice doc. Alternatively, if we don't want to keep piling guides up, we could but it out as a blog post so it's at least easily shareable whenever folks need something more in-depth

Reply:
Refactored. I removed the sections and added a "Handling Rate Limit Errors" Best Practice guide.